### PR TITLE
Bootstrap media objects for search results

### PR DESCRIPTION
**Pull request by crhallberg**
Simplify the results list templates and DOM using the [Bootstrap media object](http://getbootstrap.com/components/#media).

This will likely cause problems in custom templates.

Merge aim: Friday.

- [x] Separate favorites config
- [x] AJAX statuses
- [x] Syntax double check
- [x] Make sure checkboxes in template are responding to  configurations
- [x] Update wiki changelog with link to commit once finally flattened/merged

### CODE REVIEW

**Comment by demiankatz** [comment]
Did you consider adding a getCoverSize() method to the VuFind\View\Helper\Root\Record helper instead of adding logic to multiple controllers? I think if you refactored part of the existing getRecord() method to a separate method, you would be able to centralize all the logic in one place, and access the value directly from templates without having to modify any controllers. What do you think?

**Comment by demiankatz** [comment]
@crhallberg, there seems to be a problem with the refactored EDS template: "syntax error, unexpected 'endif' in EDS/result-list.phtml on line 24."

Also, on my test server at least, the image thumbnails have switched from the left to the right. Is this intentional and/or configurable? (I can see some advantages to it, but it's a big change if you're not expecting it).

**Comment by demiankatz** [comment]
Another round of comments:

1.) I see that the indentation in EDS/result-list.phtml is a bit out of whack. Is that an intentional result of the "minimize diffs" task, or an accident? Either way, I think it's worth fixing. I can review diffs outside of GitHub using a tool that's less sensitive to whitespace changes.

2.) Rather than using output buffering, might it make sense to use a sub-template or view helper for constructing the thumbnail portion of the output? This would be especially beneficial if we're using the exact same logic in multiple templates. (I didn't check to see if it's pure copy-and-paste here... if there's more variation, then perhaps the existing approach is indeed the best way forward).

3.) Given that we ran into EDS syntax errors, we should probably be sure we test all of the other templates too. I can help with this. Have you tested any of the non-Solr ones? Let me know if you want to split up the list somehow. I think the hardest one to test will probably be Pazpar2 -- not sure if we have easy access to any working examples of that right now. Maybe we can fake it somehow.

I'll do some more hands-on testing and visual inspection next week... but from the look of the code, this does seem to be a step in the right direction.

**Comment by crhallberg** [comment]
I fixed all the indenting, I pretty sure this is ready to go.

**Comment by crhallberg** [comment]
Right aligned thumbnails:
image

New default style:
image

File: themes/bootstrap3/less/components/search.less [hunk 1]
```diff
@@ -25,5 +52,3 @@
   font-size: inherit;
   font-style: italic;
 }
-
-.wikipedia img { margin-right: 1rem; }
```
Comment (demiankatz) [inline themes/bootstrap3/less/components/search.less]: Was the wikipedia image margin setting removed intentionally?

**Comment by demiankatz** [comment]
Some other comments:

1.) Did you know you can add `?w=1` to the end of the files diff URL on the PR to ignore whitespace? Makes reviewing a lot easier!

2.) Do we need different "thumbnail on right" settings for favorites vs. results? That's probably overcomplicating things, but I bring it up just to see what you think.

3.) I think the thumbnail looks kind of weird on the right when there are action-oriented controls immediately to the left of it. It kind of feels like maybe "thumbnail in middle" might be desirable in some scenarios. I realize that probably completely defeats the purpose of using media objects and is probably not practical -- but again, just playing devil's advocate.

4.) AJAX statuses are broken in this branch. I think it's because you've (perhaps accidentally) renamed the "ajaxItemId" class in the template to simply "ajaxItem."

5.) On a possibly related note, I also see that you have completely dropped the class on the top-level div that contains the source identifier (e.g. sourceSolr, sourceSummon, etc.). Was this intentionally removed? Are you completely certain that all code referencing this has been eliminated? (Definitely seems like something worth cleaning up now that the new lightbox is here... but I just want to be absolutely sure we don't have any legacy uses of the data).

6.) I am almost certain that the Pazpar2 template has multiple parse errors (at least two) in it related to missing colons at the ends of if statements. It would be really nice if we could come up with a way to at least fake-test this to be sure it renders properly.

7.) Does it matter that the checkbox class in list-list.phtml has now been made unconditional, so it is included even if there are no checkboxes? Was that intentional?

**Comment by crhallberg** [comment]
1.) * does happy dance* 

2.) I assumed most would want this consistent, but I don't see why not.

3.) This is the natural place for the media objects, it's also exactly what [Swarthmore](https://tripod.brynmawr.edu/find/Books/Results?lookfor=&type=AllFields) ordered.

4.) Will fix.

5.) I couldn't find any code references, and we've also added a hiddenSource to every record.

6.) I'll try copying those templates into the place of SolrDefault.

7.) Will fix.

8.) Will do.

**Comment by demiankatz** [comment]
A couple more notes:

1.) VuFind\Controller\AbstractSearch contained some code that seemed to have leaked over from the facets-modal branch; I reverted it.

2.) It looks like there is no longer a max-height on cover images (or, at least, you removed one but didn't replace it). I know there are now max-widths in place, but do we need to worry about the possibility of extremely tall/skinny thumbnails rendering strangely?

Apart from that, this is looking good to me. I'll do a bit more testing, but I think we're about ready to merge here, once you've considered whether any minor LESS adjustments are needed.

**Comment by demiankatz** [comment]
Actually, one more thing:

3.) In list view, the first entry in the list seems to be formatted incorrectly -- it is not as wide as the subsequent entries. I'll send you a screen shot on Slack.

**Comment by demiankatz** [comment]
And more issues:

4.) We seem to have lost the alternate shading of search results -- in master, you get gray backgrounds on every other result. Here, everything is white.

5.) For items without a thumbnail (for example, the authority record results), there is no padding between the result number and the title -- things look a bit squished. Can we pad this back out again? Let me know if you need me to point you to my authority index for testing purposes.

**Comment by demiankatz** [comment]
Manually merged to master for clean commit history. Thanks for all the work, Chris -- this does look subtly cleaner than before, in addition to offering new functionality and simpler markup.

**Pull request closed without merging**